onnxruntime/core/providers/cuda/math/gemm.cc

```diff
@@ -85,42 +85,42 @@ Status Gemm<T>::ComputeInternal(OpKernelContext* ctx) const {
     if (b_shape.Size() == 1) {
       // if B is (), (1,) or (1, 1), broadcast the scalar
       CUBLAS_RETURN_IF_ERROR(cublasCopyHelper(
-          Stream(ctx),
-          CublasHandle(),
-          M * N,
-          b_data,
-          0,
-          out_data,
-          1),
-                             CublasHandle(),
+                                 Stream(ctx),
+                                 GetCublasHandle(ctx),
+                                 M * N,
+                                 b_data,
+                                 0,
+                                 out_data,
+                                 1),
+                             GetCublasHandle(ctx),
                              Stream(ctx));
     } else if (b_shape.NumDimensions() == 1 || b_shape[0] == 1) {
       // B is (N,) or (1, N), broadcast using Y(N,M) = 1 * B(N,1) x ones(1,M) + 0 * Y
       CUBLAS_RETURN_IF_ERROR(cublasGemmHelper(
-          CublasHandle(),
-          CUBLAS_OP_N,
-          CUBLAS_OP_N,
-          N, M, 1,
-          /*alpha*/ &one,
-          b_data, N,
-          GetConstOnes<CudaT>(M, GetCudaStreamFromContext(ctx)), 1,
-          /*beta*/ &zero,
-          out_data, N, device_prop),
-                             CublasHandle(), 
+                                 GetCublasHandle(ctx),
+                                 CUBLAS_OP_N,
+                                 CUBLAS_OP_N,
+                                 N, M, 1,
+                                 /*alpha*/ &one,
+                                 b_data, N,
+                                 GetConstOnes<CudaT>(M, GetCudaStreamFromContext(ctx)), 1,
+                                 /*beta*/ &zero,
+                                 out_data, N, device_prop),
+                             GetCublasHandle(ctx),
                              Stream(ctx));
     } else if (b_shape.NumDimensions() == 2 && b_shape[1] == 1) {
       // B is (M, 1), broadcast using Y(N,M) = 1 * ones(N,1) x B(1,M) + 0 * Y
       CUBLAS_RETURN_IF_ERROR(cublasGemmHelper(
-          CublasHandle(),
-          CUBLAS_OP_N,
-          CUBLAS_OP_N,
-          N, M, 1,
-          /*alpha*/ &one,
-          GetConstOnes<CudaT>(N, GetCudaStreamFromContext(ctx)), N,
-          b_data, 1,
-          /*beta*/ &zero,
-          out_data, N, device_prop),
-                             CublasHandle(),
+                                 GetCublasHandle(ctx),
+                                 CUBLAS_OP_N,
+                                 CUBLAS_OP_N,
+                                 N, M, 1,
+                                 /*alpha*/ &one,
+                                 GetConstOnes<CudaT>(N, GetCudaStreamFromContext(ctx)), N,
+                                 b_data, 1,
+                                 /*beta*/ &zero,
+                                 out_data, N, device_prop),
+                             GetCublasHandle(ctx),
```

Comment:
you can change the Cublas_call macro to remove the set stream call internally, to avoid pass in stream here.